index.js

```diff
@@ -0,0 +1,114 @@
+const billNumElement = document.querySelector("#bill");
+const tipsButton = document.querySelectorAll('input[name="tip"]');
+const tipCustomElement = document.querySelector("#tip-custom");
+const numberOfPeopleElement = document.querySelector("#number-of-people");
+const numberOfPeopleErrorElement = document.querySelector(
+  "#number-of-people-error"
+);
+const resetButton = document.querySelector("#reset");
+const resultTipAmountElement = document.querySelector("#result-tip-amount");
+const resultTipTotalElement = document.querySelector("#result-total");
+
+const tips = {
+  set billNum(changeNum) {
+    this._billNum = changeNum;
+    this.tipChangeObserver();
+  },
+  set tip(changeNum) {
+    this._tip = changeNum;
+    this.tipChangeObserver();
+  },
+  set numberOfPeople(changeNumber) {
+    this._numberOfPeople = changeNumber;
+    this.tipChangeObserver();
+  },
+  set tipAcount(value) {
+    this._tipAcount = value;
+  },
+  set tipTotal(value) {
+    this._tipTotal = value;
+  },
+  get tipAcount() {
+    return this._tipAcount;
+  },
+  get tipTotal() {
+    return this._tipTotal;
+  },
+  tipChangeObserver() {
+    if (
+      !(
+        this._billNum != undefined &&
+        this._numberOfPeople != undefined &&
+        this._tip != undefined
+      )
+    ) {
+      return;
+    }
+    if (!(this._billNum > 0 && this._numberOfPeople > 0 && this._tip > 0)) {
+      return;
+    }
+    this._tipTotal = this._billNum * this._tip;
+    this._tipAcount = this._tipTotal / this._numberOfPeople;
+    changeResult(this._tipAcount, this._tipTotal);
+  },
+};
+
+const billNumChagneListner = function (event) {
+  tips.billNum = +event.target.value;
+};
+
+const tipsButtonClickListner = function (event) {
+  tips.tip = +event.target.value;
+};
```

Comment:
은연듯 든 생각인데, 이것도 뭔가 합칠 수 있을 것 같아요! (just 제안)
[입코딩]
- event.target에서 클래스 네임이나 name 속성을 가져온다
- tips라는 객체의 property의 이름을 input 태그의 name속성이나 클래스네임과 동일하게 한다
이렇게 하면 함수를 일반화해서 사용가능하지 않을까?
```suggestion
const onChangeHandler = (event) => {
  tips[event.target.name] = +event.target.value;
};
```

Reply:
오 React같은 component로 동작하는 라이브러리에서 짜면 함 생각해볼만한 것 같아요!